datadog_checks_base/datadog_checks/base/log.py

```diff
@@ -23,11 +24,45 @@ def trace(self, msg, *args, **kwargs):
             self._log(TRACE_LEVEL, msg, args, **kwargs)
 
 
+class RedactFilter(logging.Filter):
+    def __init__(self, *args, **kwargs):
+        # type: (*Any, **Any) -> None
+        super(RedactFilter, self).__init__(*args, **kwargs)
+        self.patterns = set()  # type: Set[str]
+
+    def filter(self, record):
+        # type: (logging.LogRecord) -> bool
+        record.msg = self.redact(to_native_string(record.msg))
+
+        if isinstance(record.args, dict):
+            record.args = {key: self.redact(value) for key, value in iteritems(record.args)}
+        else:
+            record.args = tuple(self.redact(arg) for arg in record.args)
+
+        return True
+
+    def redact(self, message):
+        # type: (str) -> str
+        for pattern in self.patterns:
+            message = message.replace(pattern, '********')
+        return message
+
+
 class CheckLoggingAdapter(logging.LoggerAdapter):
     def __init__(self, logger, check):
         super(CheckLoggingAdapter, self).__init__(logger, {})
         self.check = check
         self.check_id = self.check.check_id
+        self._redact_filter = RedactFilter('redact-secrets')
+        self.logger.addFilter(self._redact_filter)
```

Comment:
Let's not do this at all until first registration

Reply:
We ideally want the common case (no secret) to use the current code path (no new calls)